src/controller/artifact/annotation/v1alpha1/v1alpha1.go

```diff
@@ -0,0 +1,114 @@
+package v1alpha1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/goharbor/harbor/src/controller/artifact/annotation"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/artifact"
+	"github.com/goharbor/harbor/src/pkg/registry"
+
+	"github.com/docker/distribution/manifest/schema2"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const (
+	// V1alpha1 is the version of annotation parser
+	V1alpha1 = "v1alpha1"
+)
+
+func init() {
+	v1alpha1Parser := &v1alpha1Parser{
+		version: V1alpha1,
+		regCli:  registry.Cli,
+	}
+	if err := annotation.RegisterAnnotationParser(v1alpha1Parser, V1alpha1); err != nil {
+		log.Errorf("failed to register annotation parser for version %s: %v", V1alpha1, err)
+		return
+	}
+}
+
+type v1alpha1Parser struct {
+	version string
+	regCli  registry.Client
+}
+
+func (p *v1alpha1Parser) Parse(ctx context.Context, artifact *artifact.Artifact, manifest []byte) error {
+	if artifact.ManifestMediaType != v1.MediaTypeImageManifest && artifact.ManifestMediaType != schema2.MediaTypeManifest {
+		return nil
+	}
+	// get manifest
+	mani := &v1.Manifest{}
+	if err := json.Unmarshal(manifest, mani); err != nil {
+		return err
+	}
+	// get config layer
+	_, blob, err := p.regCli.PullBlob(artifact.RepositoryName, mani.Config.Digest.String())
+	if err != nil {
+		return err
+	}
+	defer blob.Close()
+	// parse metadata from config layer
+	metadata := map[string]interface{}{}
+	// Some artifact may not have empty config layer.
+	if mani.Config.Size != 0 {
+		if err := json.NewDecoder(blob).Decode(&metadata); err != nil {
+			return err
+		}
+	}
+	// Populate all metadata into the ExtraAttrs first.
+	artifact.ExtraAttrs = metadata
+	// parse skip-list
+	skipListAnnotationKey := fmt.Sprintf("%s.%s.%s", annotation.AnnotationPrefix, p.version, annotation.SkipList)
+	skipList, ok := mani.Config.Annotations[skipListAnnotationKey]
+	if ok {
+		skipKeyList := strings.Split(skipList, ",")
+		for _, skipKey := range skipKeyList {
+			delete(metadata, skipKey)
+		}
+		artifact.ExtraAttrs = metadata
+	}
+	// parse icon
```

Comment:
from here to end, can you just give the default icon instead of returning error? @ywk253100